source/grid/tria.cc

```diff
@@ -2095,27 +2095,34 @@ namespace internal
             // in higher dimensions, this may happen but is not
             // implemented
             if (spacedim == 2)
-              AssertThrow((n_adj_cells >= 1) && (n_adj_cells <= 2), ExcInternalError()) else AssertThrow(
-                (n_adj_cells >= 1) && (n_adj_cells <= 2),
-                ExcMessage("You have a line in your triangulation "
```

Comment:
Ugh :-(

Reply:
It looks like we forgot a semicolon after `AssertThrow` before. That really confuses clang-format.

Reply:
It was on purpose. `Assert` expands into `if (cond) {...}`, so if you want to say
```
  if (something)
    Assert (..., ...)
  else
    ...
```
you need to omit the semicolon or the `else` will no longer be syntactically correct.

Reply:
It might be interesting to see with a script if there are other places/how many other places there are where the indentation script now has things broken up in this awkward way.

Reply:
We usually put in the semicolon, though; fortunately none of our compilers are that pedantic.

Reply:
If I have some time later I'll try to put together a little script to find these. It would be good to add missing semicolons.

Reply:
What I mean to say is that you *can't* put a semicolon after the `Assert` if there is an `else` following, because then it would expand to
```
 if (something)
  {
    if (cond)
      trigger assertion
  };                 // note semicolon
  else
    ...something else...
```
Here, the `{...}` encloses a block, the semicolon is superfluous and ends an otherwise empty statement, and the `else` is now attached to that empty statement and not the `if` code block any more. That's not legal. So you *can't* have the semicolon in this situation.

Reply:
The obvious workaround, used by @masterleinad, is to enclose the `Assert` in curly braces. This way you can add the semicolon at will.